vmspawn: fix --extra-drive= option

labichn opened the pull request:
This PR fixes the vmspawn regression introduced by #36578. (See #36807.)

Not addressed: other `-drive` uses per @yuwata's comment. I'm glad to take a swing at that once the regression is fixed.

yuwata commented on src/vmspawn/vmspawn.c:
```diff
@@ -1973,6 +1973,11 @@ static int run_virtual_machine(int kvm_device_fd, int vhost_device_fd) {
+                strv_extend_many(&cmdline, "-device", "virtio-scsi-pci,id=scsi");
+        }
+
+        i = 0;
```
Please use `unsigned` and declare it here:
```c
unsigned i = 0;
STRV_FOREACH(...) {
```

yuwata reviewed (changes requested):
Please prefix the commit title with `vmspawn:`, mention the commit hash that causes the issue, and mention the issue number, something like the following:
```
vmspawn: fix --extra-drive= option

Follow-up for 369655330d57a82b6f346a77798968f135221c99.

Fixes #36807.
```

labichn commented:
Thank you for the review! I addressed your points in the latest push.

Please let me know if any of the CI failures are things I should investigate, I'm unsure whether they're related to this change. (The unit and integration tests pass locally for me.)

YHNdnzj commented on src/vmspawn/vmspawn.c:
```diff
@@ -1973,6 +1973,10 @@ static int run_virtual_machine(int kvm_device_fd, int vhost_device_fd) {
         }
 
+        if (arg_image || strv_length(arg_extra_drives) > 0)
+                strv_extend_many(&cmdline, "-device", "virtio-scsi-pci,id=scsi");
```
Please check for OOM

yuwata commented on src/vmspawn/vmspawn.c:
```diff
@@ -1984,7 +1991,15 @@ static int run_virtual_machine(int kvm_device_fd, int vhost_device_fd) {
                         return log_oom();
 
-                r = strv_extendf(&cmdline, "driver=raw,cache.direct=off,cache.no-flush=on,file.driver=file,file.filename=%s", escaped_drive);
+                r = strv_extendf(&cmdline, "driver=raw,cache.direct=off,cache.no-flush=on,file.driver=file,file.filename=%s,node-name=vmspawn_extra%u", escaped_drive, i);
```
I'd like to insert `_` before `%u`

yuwata reviewed (changes requested):
The commit hash in the commit message is wrong.
Please use `369655330d57a82b6f346a77798968f135221c99` instead.

yuwata reviewed (approved):
LGTM.
